src/radical/pilot/agent/bootstrap_0.sh

```diff
@@ -1804,7 +1804,7 @@ export PATH="$PB1_PATH"
 export LD_LIBRARY_PATH="$PB1_LDLB"
 
 # activate virtenv
-if test "$PYTHON_DIST" = "anaconda" && ! test -z $(which conda)
+if test "$PYTHON_DIST" = "anaconda" && ! test -z "\$(which conda)"
```

Comment:
I think the slash before `$` is incorrect - this will always be true:
```sh
$ which ls && echo "yes" || echo "no"
/bin/ls
yes
$ test -z $(which ls) && echo "yes" || echo "no"
no

$ which foo && echo "yes" || echo "no"
no
$ test -z $(which foo) && echo "yes" || echo "no"
yes
$ test -z "$(which foo)" && echo "yes" || echo "no"
yes
$ test -z "\$(which foo)" && echo "yes" || echo "no"
no
```

Reply:
@andre-merzky this part is what will be written into `bootstrap_2.sh`, thus to keep `$` and not execute it before to write into the script we put `\` -> `bootstrap_2.sh` will have `... test -z "$(which conda)"`